static/js/src/PurchaseModal/components/UserInfoForm/UserInfoForm.tsx

```diff
@@ -69,10 +69,72 @@ const UserInfoForm = ({ setCardValid, isGuest }: Props) => {
     return errorMessage;
   };
 
+  const { data: userInfo } = useStripeCustomerInfo();
+
+  const defaultPaymentMethod = userInfo?.customerInfo?.defaultPaymentMethod;
+
   const displayMode = (
```

Comment:
I think the solution here is too change the condition that decides if we are on `displayMode` or `editMode` rather than changing the `displayMode` component to also allow edits.

I think doing something like the following around [line 38](https://github.com/canonical/ubuntu.com/pull/12283/files#diff-ccac169c2330178b4a0ae8fbe0d3eb832a79869af42630925e71d651d2ae2786R38) should be enough to solve this issue:
```js
  const { data: userInfo } = useStripeCustomerInfo();
  const defaultPaymentMethod = userInfo?.customerInfo?.defaultPaymentMethod;
  const [isEditing, setIsEditing] = useState(isGuest || !defaultPaymentMethod);
```

Reply:
Ah, thanks @jpmartinspt! I thought we wanted to update displayMode. Fixed! 👍